Diary/Resource/Model/CoreDataManager.swift

```diff
@@ -0,0 +1,136 @@
+            }
+        })
+        return container
+    }()
+    
+    private var viewContext: NSManagedObjectContext {
+        return persistentContainer.viewContext
+    }
+    
+    private var diaryEntity: NSEntityDescription? {
+        return NSEntityDescription.entity(forEntityName: CoreDataNamespace.diary, in: viewContext)
+    }
+    
+    private func saveContext() {
+        let context = persistentContainer.viewContext
+        if context.hasChanges {
+            do {
+                try context.save()
+            } catch {
+                AlertManager.shared.sendError(title: ErrorNamespace.saveError)
+            }
+        }
+    }
+    
+    func insertDiary(_ diaryModel: DiaryModel?) {
+        guard let diaryModel = diaryModel,
+              let diaryEntity = diaryEntity else { return }
+        
+        let managedObject = NSManagedObject(entity: diaryEntity, insertInto: viewContext)
+        managedObject.setValue(diaryModel.title, forKey: CoreDataNamespace.title)
+        managedObject.setValue(diaryModel.body, forKey: CoreDataNamespace.body)
+        managedObject.setValue(diaryModel.createdAt, forKey: CoreDataNamespace.createAt)
+        saveContext()
+    }
+    
+    func fetchDiary(with id: NSManagedObjectID?) -> Diary? {
+        guard let id = id else { return nil }
+        
+        guard let object = viewContext.object(with: id) as? Diary else { return nil }
+        
+        return object
+    }
+    
+    func fetchAllDiaries() -> [Diary] {
+        do {
+            let request = Diary.fetchRequest()
+            let results = try viewContext.fetch(request)
+            
+            return results
+        } catch {
+            AlertManager.shared.sendError(title: ErrorNamespace.loadingFailure)
+        }
+        
+        return []
+    }
+    
+    func fetchAllDiaryModels() -> [DiaryModel] {
+        var diaryModels: [DiaryModel] = []
+        let fetchedResults = fetchAllDiaries()
+        
+        for result in fetchedResults {
+            let diary = DiaryModel(id: result.objectID,
+                                   title: result.title ?? Namespace.empty,
+                                   body: result.body ?? Namespace.empty,
+                                   createdAt: result.createdAt ?? Date())
+            diaryModels.append(diary)
+        }
+        
+        return diaryModels
+    }
+    
+    func fetchID(of diaryModel: DiaryModel?) -> NSManagedObjectID? {
+        guard let diaryModel = diaryModel else { return nil }
+        
+        let fetchRequest = Diary.fetchRequest()
+        fetchRequest.predicate = NSPredicate(format: CoreDataNamespace.regex,
+                                             argumentArray: [diaryModel.title, diaryModel.body, diaryModel.createdAt])
+        
+        let result = try? viewContext.fetch(fetchRequest)
+        if result?.first?.objectID == nil {
+            return nil
+        } else {
+            return result?.first?.objectID
+        }
```

Comment:
```suggestion
    return result?.first?.objectID
```
굳이 옵셔널 체크를 할 필요는 없어보여요. 꼭 옵셔널 바인딩한 값이 필요한게 아니라면 저는 안하는 편입니다.